### PR TITLE
Feat/okex v5 api migration

### PR DESCRIPTION
**Pull request by keithbaum**
Migration of okex v3 api to v5

**Before submitting this PR, please make sure**:

- [ ] Your code builds clean without any errors or warnings
- [ ] You are using approved title ("feat/", "fix/", "docs/", "refactor/")

**A description of the changes proposed in the pull request**:

### CODE REVIEW

**Comment by keithbaum** [comment]
Connector as is, cannot place orders because client order id convention for v5 does not allow first character to be a number

image

Whereas for v3 convention was different

image


This is the error in the API request response
image

**Comment by dennisocana** [comment]
Changes status to draft as this is not yet ready for QA testing. Reached out to OKEx team for the `CLIENT_ID_PREFIX` since it does not meet their v5 requirements.

**Comment by keithbaum** [comment]
As an additional test, tried changing the CLIENT_ID_PREFIX to start with a letter instead, and connector worked without problems.

**Comment by keithbaum** [comment]
Just tested with the original client_id_prefix and it is now placing orders successfully. This change can now be reviewed.

**Comment by rxlxrxsx** [comment]
Getting this error every 10 seconds after starting the strategy, it seems to only happen when there's no balance on one side of the market pair.

```
2021-04-19 14:02:29,097 - 8884 - hummingbot.connector.exchange.okex.okex_exchange - ERROR - Unexpected error in user stream listener lopp. list index out of range
Traceback (most recent call last):
  File "hummingbot/connector/exchange/okex/okex_exchange.pyx", line 538, in hummingbot.connector.exchange.okex.okex_exchange.OkexExchange._user_stream_event_listener
    details = data["details"][0]
IndexError: list index out of range
```

Screen Shot 2021-04-19 at 2 03 11 PM

Logs and config: 
[okex_v5.zip](https://github.com/CoinAlpha/hummingbot/files/6333618/okex_v5.zip)

**Comment by PtrckM** [comment]
unit testing no error when using `export MOCK_API_ENABLE=true` when disabled throws some errors
  - `test_okex_auth.py` still using v3 https://github.com/CoinAlpha/hummingbot/blob/feat/okex_v5_api/test/integration/test_okex_auth.py#L19 but no error when running the unit test
     
```
(hummingbot) ubuntu@ip-172-31-27-100:~/okex/test/integration$ nosetests -v -s test_okex_auth.py
test_aut_withouth_params (test.integration.test_okex_auth.TestAuth) ... ok
```

  - `test_okex_api_order_book_data_source.py` failing on  
```
test_example_market (test.integration.test_okex_api_order_book_data_source.TestOkexAPIOrderBookDataSource) ... ok
test_get_last_traded_prices (test.integration.test_okex_api_order_book_data_source.TestOkexAPIOrderBookDataSource) ... ok
test_get_snapshot (test.integration.test_okex_api_order_book_data_source.TestOkexAPIOrderBookDataSource) ... ok
test_listen_for_order_book_diffs (test.integration.test_okex_api_order_book_data_source.TestOkexAPIOrderBookDataSource) ... ok
test_listen_for_trades (test.integration.test_okex_api_order_book_data_source.TestOkexAPIOrderBookDataSource) ... FAIL

======================================================================
FAIL: test_listen_for_trades (test.integration.test_okex_api_order_book_data_source.TestOkexAPIOrderBookDataSource)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/ubuntu/okex/test/integration/test_okex_api_order_book_data_source.py", line 101, in test_listen_for_trades
    asyncio.get_event_loop().run_until_complete(self.listen_for_trades())
  File "/home/ubuntu/miniconda3/envs/hummingbot/lib/python3.8/asyncio/base_events.py", line 616, in run_until_complete
    return future.result()
  File "/home/ubuntu/okex/test/integration/test_okex_api_order_book_data_source.py", line 96, in listen_for_trades
    self.assertFalse(q.empty())
AssertionError: True is not false
```

   - `test_okex_market.py` still using v3 on https://github.com/CoinAlpha/hummingbot/blob/feat/okex_v5_api/test/integration/test_okex_market.py#L116 and https://github.com/CoinAlpha/hummingbot/blob/feat/okex_v5_api/test/integration/test_okex_market.py#L118

**Review by RC-13 — APPROVED** [review]
Tested and no additional issue found by QA. As per discussion during client stand-up, we can approve this PR and we can have a separate PR for updating the unit test.

File: hummingbot/connector/exchange/okex/okex_exchange.pyx [hunk 1]
```diff
@@ -1026,13 +992,20 @@ cdef class OkexExchange(ExchangeBase):
...
-                amount: Decimal,
-                price: Decimal = s_decimal_NaN) -> TradeFee:
+    async def get_fee(self,
+                      base_currency: str,
+                      quote_currency: str,
+                      order_type: OrderType,
+                      order_side: TradeType,
+                      amount: Decimal,
+                      price: Decimal = s_decimal_NaN) -> TradeFee:
+        msg = await self._api_request("GET",
```
Comment (Nullably) [inline hummingbot/connector/exchange/okex/okex_exchange.pyx:1002]: Not a good idea to make API call to get fee data here. This get_fee is called many times and often in a loop, those caller functions will then take much longer time to complete.
It is ok to use the original implementation where the fee is estimated.

**Review by Nullably — CHANGES_REQUESTED** [review]

File: hummingbot/connector/exchange/okex/okex_exchange.pyx (hunk 1 above)
Comment (keithbaum) [reply hummingbot/connector/exchange/okex/okex_exchange.pyx]: Latest commit pushed back to original get_fee estimation method

**Review by rxlxrxsx — APPROVED** [review]
Tested and confirmed working by QA.

**Review by Nullably — APPROVED** [review]
LGTM

**Pull request merged**